packages/meditrak-server/src/routes/utilities/SurveyResponseVariablesExtractor.js

```diff
@@ -0,0 +1,144 @@
+    return { surveyResponse, surveyId, country };
+  }
+
+  async getSurveys(surveyId, surveyCodes, countryId) {
+    let surveys;
+    if (surveyId) {
+      // A surveyId of interest passed in, only export that
+      surveys = [await this.models.survey.findById(surveyId)];
+    } else if (surveyCodes) {
+      const surveyFindConditions = {
+        // surveyCodes may be passed through as a comma separated string or as an array, which looks like this in a query:
+        // ?surveyCodes=code1&surveyCodes=code2
+        code: {
+          comparisonType: 'whereIn',
+          args: Array.isArray(surveyCodes) ? [surveyCodes] : [surveyCodes.split(',')],
+        },
+      };
+      if (countryId) {
+        // Fetch surveys where country_ids is empty (enabled in all countries) or contains countryId
+        // eslint-disable-next-line no-underscore-dangle
+        surveyFindConditions._and_ = {
+          country_ids: '{}',
+          _or_: {
+            country_ids: { comparator: '@>', comparisonValue: [countryId] },
+          },
+        };
+      }
+      surveys = await this.models.survey.find(surveyFindConditions);
+    } else {
+      // No specific surveyId passed in, so export all surveys that apply to this country
+      const allSurveys = await this.models.survey.all();
+      surveys = allSurveys.filter(
+        survey => survey.country_ids.length === 0 || survey.country_ids.includes(countryId),
+      );
+    }
+    if (!surveys || surveys.length < 1) {
+      throw new ValidationError('Survey not found. Please check permissions');
+    }
+
+    return surveys;
+  }
+
+  async getParametersFromInput(countryCode, entityCode, countryId, entityIds, surveyResponseId) {
+    let country;
+    let entities;
+    let surveyId;
+    let surveyResponse;
+    if (countryCode) {
+      const variables = await this.getVariablesByCountryCode(countryCode);
+      country = variables.country;
+      countryId = variables.countryId;
+    }
+
+    if (entityCode) {
+      const variables = await this.getVariablesByEntityCode(entityCode);
+      country = variables.country;
+      entities = variables.entities;
+    } else if (countryId) {
+      const variables = await this.getVariablesByCountryId(countryId);
+      country = variables.country;
+      entities = variables.entities;
+    } else if (entityIds) {
+      const variables = await this.getVariablesByEntityIds(entityIds);
+      country = variables.country;
+      countryId = variables.countryId;
+      entities = variables.entities;
+    } else if (surveyResponseId) {
+      const variables = await this.getVariablesBySurveyResponseId(surveyResponseId);
+      country = variables.country;
+      surveyId = variables.surveyId;
+      surveyResponse = variables.surveyResponse;
+    } else {
+      throw new ValidationError(
+        'Please specify either surveyResponseId, countryId, countryCode, facilityCode or entityIds',
+      );
+    }
```

Comment:
```suggestion  
  async getParametersFromInput(countryCode, entityCode, countryId, entityIds, surveyResponseId){
    if (countryCode) return this.getVariablesByCountryCode(countryCode);
    if (entityCode) return this.getVariablesByEntityCode(entityCode);
    // ...
    throw new ValidationError(
      'Please specify either surveyResponseId, countryId, countryCode, facilityCode or entityIds',
    );
  }
```
Could we shorten this a bit?

Reply:
Nice! Would love to change to your approach.